swig/ChunkedArray.hpp

```diff
@@ -0,0 +1,240 @@
+#ifndef __CHUNKED_ARRAY_H__
+#define __CHUNKED_ARRAY_H__
+
+#include <new>
+#include <vector>
+#include <algorithm>
+
+#include <stdint.h>
+#include <assert.h>
+
+/**
+ * Container that manages a dynamic array of fixed-length chunks.
+ *
+ * The class also takes care of allocation & release of the underlying
+ * memory. It can be used with either a high or low-level API.
+ *
+ * The high-level API allocates chunks as needed, manages addresses automatically and keeps
+ * track of number of inserted elements, but is not thread-safe (ok as usually input is a streaming iterator).
+ * For parallel input sources the low-level API must be used.
+ *
+ * Note: When using this for `LGBM_DatasetCreateFromMats` use a
```

Comment:
mentioned this on the issue as well - can we add sparse support too, and what would that involve?

Reply:
Answered on that issue :)